src/RawData/storageController.ts

```diff
@@ -86,37 +90,52 @@ const createStreamFromChunks = async (filePath: string) => {
 const splitRawDataStreamIntoParameters = async (
   fileStream: stream.PassThrough,
   fileId: string,
-  filePath: string
+  filePath: string,
+  selectedBuckets: boolean[]
 ) =>
   new Promise((resolve, reject) => {
     const outstandingFunctions: (() => Promise<void>)[] = [];
     fileStream
       .pipe(replacestream(/:\s*Infinity/g,':null'))
       .pipe(JSONStream.parse(["posterior", "content", { emitKey: true }]))
       .on("data", async (data: { key: string; value: any[] }) => {
-        // TODO: Pause stream until processed
-        const saveParameter = async () => {
-          const parameterId = uuidv4();
-          const filepath = path.join(
-            process.env.DATA_PATH,
-            PROCESSED_FOLDER,
-            fileId,
-            parameterId + ".json"
-          );
-          await databasePool.query(INSERT_BASE_PARAMETER, [
-            parameterId,
-            data.key,
-            fileId
-          ]);
-          await writeFile(filepath, JSON.stringify(data.value), {
-            flag: "w+"
-          });
-        };
-        outstandingFunctions.push(saveParameter);
+
+        if ((selectedBuckets[0] == true && intrinsicParameters.includes(data.key)) ||
+        (selectedBuckets[1] == true && extrinsicParameters.includes(data.key)) ||
+        ((selectedBuckets[2] == true && (
+            !(intrinsicParameters.includes(data.key))
+            && !(extrinsicParameters.includes(data.key))
+          ))
+        )) {
```

Comment:
Can we add a check here so that parameters with non-number (or complex numbered) values don't get saved?

Sample problem data:
![image](https://user-images.githubusercontent.com/101340211/195031741-a075dea3-abbe-41e5-b22c-b4b943fc1119.png)

Reply:
Fixed this issue with latest commit